src/FakeItEasy.Analyzer/NonVirtualSetupAnalyzer.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace FakeItEasy.Analyzer
+{
+    using System;
+    using System.Collections.Immutable;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.Diagnostics;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    class NonVirtualSetupAnalyzer:DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = CreateSupportedDiagnostics();
+
+        public override void Initialize(AnalysisContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            context.RegisterSyntaxNodeAction(AnalyzeCall, SyntaxKind.InvocationExpression);
+        }
+
+        private static void AnalyzeCall(SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpression = (InvocationExpressionSyntax)context.Node;
+
+            if (IsInterface(context, invocationExpression))
+                return;
+
+            if (NotSetupCall(invocationExpression))
+                return;
+
+            var symbolInfo = context.SemanticModel.GetSymbolInfo(context.Node);
+
+            if (!symbolInfo.Symbol.IsVirtual)
+            {
+                var location = invocationExpression.GetLocation();
+
+                Diagnostic diagnostic = Diagnostic.Create(DiagnosticDefinitions.NonVirtualSetup, location, symbolInfo.Symbol.Name);
+                context.ReportDiagnostic(diagnostic);
+            }
+        }
+
+        private static bool NotSetupCall(InvocationExpressionSyntax invocationExpression)

```

Comment:
I also prefer the affirmative form; `if (!IsSetupCall(invocationExpression))` would be better IMO.